util/predictoor/query.py

```diff
@@ -0,0 +1,97 @@
+from typing import Dict, Optional, Tuple
+
+from util.graphutil import submitQuery
+from util.predictoor.models import Predictoor, Prediction
+from util.constants import DEPLOYER_ADDRS
+from util.networkutil import DEV_CHAINID
+
+
+def queryPredictoors(
+    st_block: int, end_block: int, chainID: int
+) -> Dict[str, Predictoor]:
+    """
+    @description
+        Queries the predictPredictions GraphQL endpoint for a given
+        range of blocks and chain ID, and returns a dictionary of
+        Predictoor objects, where each Predictoor object represents
+        a unique user who has made predictions during the given block range.
+
+    @params
+        st_block (int) -- The starting block number for the query.
+        end_block (int) -- The ending block number for the query.
+        chainID (int) -- The ID of the chain to query.
+
+    @return
+        predictoors -- A dictionary of address to Predictoor objects
+
+    @raises
+        AssertionError: If the result of the query contains an error.
+    """
+    predictoors: Dict[str, Predictoor] = {}
+
+    chunk_size = 1000
+    offset = 0
+
+    while True:
+        # pylint: disable=line-too-long
+        query = """
+        {
+            predictPredictions(where: {block_gte:%s, block_lte:%s}, skip:%s, first:%s) {
+                id,
+                slot{
+                    status,
+                    predictContract {
+                        id
+                        token {
+                            nft {
+                                owner {
+                                    id
+                                }
+                            }
+                        }
+                    }
+                    slot
+                },
+                user {
+                    id
+                }
+                payout
+                block
+            }
+        }
+        """ % (
+            st_block,
+            end_block,
+            offset,
+            chunk_size,
+        )
+        offset += chunk_size
+        result = submitQuery(query, chainID)
+        if "error" in result:
+            raise AssertionError(result)
+        if "data" not in result:
+            raise AssertionError(result)
+
+        predictions = result["data"]["predictPredictions"]
+        if len(predictions) == 0:
+            break
+
+        for prediction_dict in predictions:
+            owner = prediction_dict["slot"]["predictContract"]["token"]["nft"]["owner"]
+            if chainID != DEV_CHAINID:
```

Comment:
I think you can combine these ifs with an or.
Also, I would suggest handling these assertions in from_query_result. I.e. try creating the object (no adding it to the list yet). Catch any exceptions from its creation and "continue" if an Exception is caught. In this way you avoid processing these dicts in two places.